spec/requests/medications_spec.rb

```diff
@@ -144,7 +95,7 @@ def post_create(medication_params)
           it 'has no validation errors and has correct refill' do
             post_create valid_medication_params
             expect(assigns(:medication).errors).to be_empty
-            expect(assigns(:medication)[:refill].strftime('%d/%m/%Y')).to eq('01/01/2020')
+            expect(response.body).to include(valid_medication_params.refill)
```

Comment:
Is this sufficient by itself (i.e. if we just drop the line above to be rid of `assigns`)? Is there a more complete expected body we could use with `eq` here instead that captures the lack of validation errors?

Reply:
Technically ya. The "it" is specifically saying there is no validation errors, which that second expectation isn't. I'd say you're fine just removing

Reply:
On a related note:

https://www.betterspecs.org/#single

Reply:
Looking at the controller itself, wouldn't response be something like

```ruby
expect(response).to have_http_status(302)
```

Maybe we can replace the assigns with that or remove the assigns and put that assertion. If we remove the assigns, let's just called the test:

```ruby
it 'has correct refill' do
```

otherwise with the redirect check

```ruby
it 'redirects to medication and has correct refill' do
```

----

If it did have an error, wouldn't it be `have_http_status(422)` for unprocessable entity?

I'm basing on the condition below in the **create** method:

```ruby
if @medication.save
  redirect_to_medication(@medication)
else
  render_unprocessable_medication
end
```